src/services/conversations/conversations.hooks.js

```diff
@@ -0,0 +1,165 @@
+import logger from 'winston';
+import commons from 'feathers-hooks-common';
+import { updatedAt, createdAt } from '../../hooks/timestamps';
+import { disallow } from 'feathers-hooks-common';
+import sanitizeAddress from '../../hooks/sanitizeAddress';
+import errors from 'feathers-errors';
+import sanitizeHtml from '../../hooks/sanitizeHtml';
+
+/**
+  API for creating conversations
+  The following params are accepted when creating a message for a conversation
+
+  @params:
+    milestoneId:        (string) the id of the milestone that this conversation belongs to
+    messageContext:     (string) context of the message, see MESSAGE_CONTEXT below
+    message:            (string) the actual message
+    replyToId:          (string) id of the message that this is a reply to
+
+  @param ownerAddress (string) is automatically set based on the current logged in user
+**/
+
+
+/**
+  Available conversation types. This roughly follows the milestone status
+  replyTo is for threaded messages
+**/
+const MESSAGE_CONTEXT = [
+  'proposed',
+  'rejected',
+  'NeedsReview',
+  'Completed',
+  'Canceled',
+  'replyTo',
+  'proposedRejected',
+  'proposedAccepted',
+  'rePropose'
+];
+
+/**
+ Only people involved with the milestone can create conversation
+ Sets the address creating the conversation as the owner
+ */
+const restrictAndSetOwner = () => context => {
+  const { app, params } = context;
+  const { milestoneId } = context.data;
+  let { user } = params;
+
+  // external calls need a user
+  if (!user && context.params.provider) throw new errors.NotAuthenticated();
+
+  return new Promise((resolve, reject) => {
```

Comment:
are you wrapping in a promise here so the error can be logged & then reject again? Would it be cleaner to just `throw new Error();` in the catch statement?

Reply:
Perhaps I'm confused, but I thought a hook needs to return a promise if it queries another dataset.

Reply:
right, but `app.service.get` returns a promise already, so no need to wrap with another